File: backend/pokerogue/src/main/java/com/pokerogue/helper/pokemon/dto/PokemonResponse.java
package com.pokerogue.helper.pokemon.dto;

import com.pokerogue.helper.global.config.ImageUrl;
import com.pokerogue.helper.pokemon.data.Pokemon;
import com.pokerogue.helper.type.dto.PokemonTypeResponse;
import java.util.List;

public record PokemonResponse(
        String id,
        Long pokedexNumber,
        String name,
        String formName,
        String image,
        String backImage,
        List<PokemonTypeResponse> pokemonTypeResponse,
        Integer generation,
        Integer totalStats,
        Integer hp,
        Integer speed,
        Integer attack,
        Integer defense,
        Integer specialAttack,
        Integer specialDefense
) {
    public static PokemonResponse from(Pokemon pokemon, List<PokemonTypeResponse> pokemonTypeResponses) {
        return new PokemonResponse(
                pokemon.getId(),
                (long) pokemon.getPokedexNumber(),
                pokemon.getName(),
                pokemon.getFormName(),
                ImageUrl.getPokemonImage(pokemon.getImageId()), //image front
                ImageUrl.getPokemonBackImage(pokemon.getImageId()), //back
                pokemonTypeResponses,
                pokemon.getGeneration(),
                pokemon.getBaseTotal(),
                pokemon.getHp(),
                pokemon.getAttack(),
                pokemon.getDefense(),
                pokemon.getSpecialAttack(),
                pokemon.getSpecialDefense(),
                pokemon.getSpeed()
        );
    }
}

File: android/app/src/main/java/poke/rogue/helper/remote/dto/response/pokemon/PokemonResponse.kt
package poke.rogue.helper.remote.dto.response.pokemon

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import poke.rogue.helper.remote.dto.response.type.PokemonTypeResponse

@Serializable
data class PokemonResponse(
    val id: Long,
    val pokedexNumber: Long,
    @SerialName("koName")
    val name: String,
    val image: String,
    @SerialName("pokemonTypeResponses")
    val types: List<PokemonTypeResponse>,
)
